elastalert/test_rule.py

```diff
@@ -23,63 +36,244 @@ def print_terms(terms, parent):
             print_terms(terms[term], parent + term + '.')
 
 
-def check_files():
-    print("Note: This tool is for testing filters and config syntax. It will not process data or alert.\n")
-    parser = argparse.ArgumentParser(description='Validate a rule configuration')
-    parser.add_argument('files', metavar='file', type=str, nargs='+', help='rule configuration filename')
-    parser.add_argument('--schema-only', action='store_true', help='Show only schema errors; do not run query')
-    parser.add_argument('--days', type=int, default=[1, 7], nargs='+', help='Query the previous N days with this rule')
-    args = parser.parse_args()
+def check_files(args):
+    """ Loads a rule config file, performs a query over the last day (args.days), lists available keys
+    and prints the number of results. """
+    filename = args.file
+    with open(filename) as fh:
+        conf = yaml.load(fh)
+    load_options(conf)
+    print("Successfully loaded %s\n" % (conf['name']))
+
+    if args.schema_only:
+        return []
+
+    # Set up elasticsearch client and query
+    es_client = Elasticsearch(host=conf['es_host'], port=conf['es_port'])
+    start_time = ts_now() - datetime.timedelta(days=args.days)
+    end_time = ts_now()
+    ts = conf.get('timestamp_field', '@timestamp')
+    query = ElastAlerter.get_query(conf['filter'], starttime=start_time, endtime=end_time, timestamp_field=ts)
+    index = ElastAlerter.get_index(conf, start_time, end_time)
+
+    # Get one document for schema
+    try:
+        res = es_client.search(index, size=1, body=query, ignore_unavailable=True)
+    except Exception as e:
+        print("Error running your filter:")
+        print(repr(e)[:2048])
+        return None
+    num_hits = len(res['hits']['hits'])
+    if not num_hits:
+        return []
+
+    terms = res['hits']['hits'][0]['_source']
+    doc_type = res['hits']['hits'][0]['_type']
+
+    # Get a count of all docs
+    count_query = ElastAlerter.get_query(conf['filter'], starttime=start_time, endtime=end_time, timestamp_field=ts, sort=False)
+    count_query = {'query': {'filtered': count_query}}
+    try:
+        res = es_client.count(index, doc_type=doc_type, body=count_query, ignore_unavailable=True)
+    except Exception as e:
+        print("Error querying Elasticsearch:")
+        print(repr(e)[:2048])
+        return None
+
+    num_hits = res['count']
+    print("Got %s hits from the last %s day%s" % (num_hits, args.days, 's' if args.days > 1 else ''))
+    print_terms(terms, '')
+
+    # Check for missing keys
+    pk = conf.get('primary_key')
+    ck = conf.get('compare_key')
+    if pk and not lookup_es_key(terms, pk):
+        print("Warning: primary key %s is either missing or null!")
+    if ck and not lookup_es_key(terms, ck):
+        print("Warning: compare key %s is either missing or null!")
```

Comment:
Perhaps logging.info is more appropriate here? (And in line with my concerns about printing errors on stderr, we could use logging.error in those cases for consistency.)